src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js

```diff
@@ -68,27 +68,45 @@ class MapViewVisualization extends Visualization {
         }
 
         let defaultMapStyle = 'mapbox://styles/mapbox/light-v9';
+        let accentColor     = [1,234,146];
+        let defaultRadius   = 150;
         if (document.getElementById("root").classList.contains("dark")){
             defaultMapStyle = 'mapbox://styles/mapbox/dark-v9';
+            accentColor     = [222,162,47];
         }
 
-
         const deckgl = new deck.DeckGL({
             container: 'map',
             mapboxApiAccessToken: 'pk.eyJ1IjoiZW5zby1vcmciLCJhIjoiY2tmNnh5MXh2MGlyOTJ5cWdubnFxbXo4ZSJ9.3KdAcCiiXJcSM18nwk09-Q',
             mapStyle: parsedData.mapStyle || defaultMapStyle,
             initialViewState: {
                 longitude: parsedData.longitude || 0.0,
                 latitude: parsedData.latitude || 0.0,
-                zoom: parsedData.zoom || 3,
+                zoom: parsedData.zoom || 11,
                 pitch: parsedData.pitch || 0
             },
             controller: parsedData.controller || true
         });
 
-        deckgl.setProps({
-            layers: parsedData.layers || []
-        });
+        let preparedDP = []
+        if (parsedData.type === "GeoPoint") {
+            preparedDP.push({position:[parsedData.longitude,parsedData.latitude], color:parsedData.color || accentColor, radius:parsedData.radius || defaultRadius});
+        } else {
+            let dataPoints = parsedData.layers[0].data || []
+            dataPoints.forEach(x => preparedDP.push({position:[x.longitude,x.latitude], color:x.color || accentColor, radius:x.radius || defaultRadius}));
+        }
+
+        const scatterplotLayer = new deck.ScatterplotLayer({
+            data: preparedDP,
+            getFillColor: d => d.color,
+            getRadius: d => d.radius
+        })
+
+        if (parsedData.type === "GeoPoint" || parsedData.layers[0].type === "ScatterplotLayer") {
+            deckgl.setProps({
+                layers: [scatterplotLayer]
+            });
+        }
     }
```

Comment:
Ok, Maciek, we need a much higher quality code here. I'm sorry to write this, but this is WAY below your standard code quality. A lot of big problems:
1. It does not implement what we've been talking about - In case of missing long / lat / zoom, it sohuld compute it based on the points in ScatterPlotLayer.
2. The lines are way too long
3. `parsedData.layers[0].data` this code will crash at runtime if layers was empty (!!!)
4. In case multiple layers are provided, and ScatterPlot was not the first layer, it will get ignored. I'm ok with ignoring layers we do not support, but why scatterplot has to be the first layer?
5. Strings like "GeoPoint" should be refactored to top of the file to some constant sections - they will be changed together with Enso types, they need to be easily accessible.
6. This code has performance problems - it creates `new deck.ScatterplotLayer` even if it is not using it later - it doesnt make sense.
7. The docs are not updated - the docs should contain good description of the input format with an example.
8. This doesnt support optional fields, like longtitude or latiitude - we were talking about that yesterday, that these fields are `Option<Number>` in Enso. (None is translated to null in JSON).
9. Name of the variables - `preparedDP`, what is `DP`? 
10. These are wrong `x.radius || defaultRadius` - why you want to use default radius if someone sets the radius of 0? For example to hide that point?